src/Middleware/Rewrite/src/UrlActions/RedirectAction.cs

```diff
@@ -36,7 +36,7 @@ internal class RedirectAction : UrlAction
                 statusCode,
                 pattern,
                 queryStringAppend,
-                queryStringDelete: true,
+                queryStringDelete: !queryStringAppend,
```

Comment:
Looking at this a little more, I now see that UrlRewriteFileParser creates RedirectActions with queryStringAppend [defaulted to true]( https://github.com/dotnet/aspnetcore/blob/cc1f23c5f8afb7d2a00405f19811d2372c4fcec2/src/Middleware/Rewrite/src/IISUrlRewrite/UrlRewriteFileParser.cs#L213).

The rule in VerifyTrackAllCaptures is one that is pulling data out of the query string and putting it into the path. Appending the query string in this case would be unexpected and breaking.

We have to decide how the UrlRewriteFileParser should behave when AppendQueryString is unspecified. Right now, it looks like it appends the request query string if there's a querystring in the pattern, and does not append the request querystring if there's no querystring in the pattern. I doubt this is how IIS behaves. @jkotalik Do you know what the behavior is supposed to be?

Reply:
I tested with IIS, it appends the query string every time unless appendQueryString="false"

Reply:
Yeah, that makes sense.

Reply:
Could we go ahead and remove this constructor and fix the call site in UrlRewriteFileParser to pass in queryStringDelete as !queryStringAppend.

I realize a lot of this confusion is coming from the fact that this class is shared between UrlRewrite and Apache ModRewrite.